hoomd/md/update.py

```diff
@@ -149,133 +150,134 @@ class mueller_plathe_flow:
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_updater = _md.MuellerPlatheFlow(
-                hoomd.context.current.system_definition, group.cpp_group,
-                flow_target.cpp_variant, slab_direction, flow_direction,
-                n_slabs, min_slab, max_slab)
+    def _attach(self):
+        group = self._simulation.state._get_group(self.filter)
+        sys_def = self._simulation.state._cpp_sys_def
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            self._cpp_obj = _md.MuellerPlatheFlow(
+                sys_def, group, self.flow_target, self.slab_direction,
+                self.flow_direction, self.n_slabs, self.min_slab, self.max_slab,
+                self.flow_epsilon)
         else:
-            self.cpp_updater = _md.MuellerPlatheFlowGPU(
-                hoomd.context.current.system_definition, group.cpp_group,
-                flow_target.cpp_variant, slab_direction, flow_direction,
-                n_slabs, min_slab, max_slab)
-
-        self.setupUpdater(period)
-
-    def get_n_slabs(self):
-        """Get the number of slabs."""
-        return self.cpp_updater.getNSlabs()
-
-    def get_min_slab(self):
-        """Get the slab id of min velocity search."""
-        return self.cpp_updater.getMinSlab()
-
-    def get_max_slab(self):
-        """Get the slab id of max velocity search."""
-        return self.cpp_updater.getMaxSlab()
-
-    def get_flow_epsilon(self):
-        """Get the tolerance between target flow and actual achieved flow."""
-        return self.cpp_updater.getFlowEpsilon()
-
-    def set_flow_epsilon(self, epsilon):
-        """Set the tolerance between target flow and actual achieved flow.
-
-        Args:
-            epsilon (float): New tolerance for the deviation of actual and
-                achieved flow.
-
-        """
-        return self.cpp_updater.setFlowEpsilon(float(epsilon))
-
-    def get_summed_exchanged_momentum(self):
-        """Returned the summed up exchanged velocity of the full simulation."""
-        return self.cpp_updater.getSummedExchangedMomentum()
-
-    def has_min_slab(self):
-        """Returns, whether this MPI instance is part of the min slab."""
-        return self.cpp_updater.hasMinSlab()
-
-    def has_max_slab(self):
-        """Returns, whether this MPI instance is part of the max slab."""
-        return self.cpp_updater.hasMaxSlab()
-
-    X = _md.MuellerPlatheFlow.Direction.X
-    """Direction Enum X for this class"""
-
-    Y = _md.MuellerPlatheFlow.Direction.Y
-    """Direction Enum Y for this class"""
+            self._cpp_obj = _md.MuellerPlatheFlowGPU(
+                sys_def, group, self.flow_target, self.slab_direction,
+                self.flow_direction, self.n_slabs, self.min_slab, self.max_slab,
+                self.flow_epsilon)
+        super()._attach()
 
-    Z = _md.MuellerPlatheFlow.Direction.Z
-    """Direction Enum Z for this class"""
+    @property
```

Comment:
Should this be loggable? If so, we should use `hoomd.logging.log` also, this should use a `DataAccessError` not return `None`. If using `hoomd.logging.log` this should be done by setting the `requires_run` flag to true.

Reply:
I agree that this should be loggable. Then we don't have to worry about  `self._attached` here and let do `hoomd.logging.log` its job.